dali/pipeline/data/tensor_vector.cc

```diff
@@ -149,8 +149,10 @@ void TensorVector<Backend>::SetSize(int new_size) {
 
 
 template <typename Backend>
-void TensorVector<Backend>::set_type(const TypeInfo &new_type) {
-  type_ = new_type;
+void TensorVector<Backend>::set_type(DALIDataType new_type) {
+  if (type_.id() == new_type)
+    return;
+  type_ = TypeTable::GetTypeInfo(new_type);
```

Comment:
Just a nitpick, but the tl_->set_type checks for valid type, so maybe, we should:
```suggestion
  tl_->set_type(new_type);
  for (auto t : tensors_) {
    t->set_type(new_type);
  }
  type_ = tl_->type_info();
```


This also works slightly differently if we have NoType, and try to set it - it will be a no op - but it was already broken a bit, so we probably don't need to mix in those changes here.

Reply:
If anything, it is _less_ broken now :)

Reply:
Fixed.